tests/testsuite/jobserver.rs

```diff
@@ -212,33 +215,104 @@ test-runner:
         .env("CARGO", cargo_exe())
         .arg("test-runner")
         .arg("-j2")
-        .with_stderr_contains("[..]this is a runner[..]")
+        .with_stderr_data(str![[r#"
+[FINISHED] `test` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+[RUNNING] unittests src/lib.rs (target/debug/deps/cargo_jobserver_check-[HASH][EXE])
+this is a runner
+
+"#]])
         .run();
 
     // but not from `-j` flag
     p.cargo("run -j2")
         .with_status(101)
-        .with_stderr_contains("[..]no jobserver from env[..]")
+        .with_stderr_data(str![[r#"
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+[RUNNING] `target/debug/cargo-jobserver-check[EXE]`
+thread 'main' panicked at src/main.rs:5:43:
+no jobserver from env: NotPresent
+[NOTE] run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+...
+
+"#]])
         .run();
     p.cargo("run -j2")
         .env("PATH", path)
         .arg("--config")
         .arg(config_value)
         .with_status(101)
-        .with_stderr_contains("[..]this is a runner[..]")
-        .with_stderr_contains("[..]no jobserver from env[..]")
+        .with_stderr_data(str![[r#"
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+[RUNNING] `runner target/debug/cargo-jobserver-check[EXE]`
+this is a runner
+thread 'main' panicked at src/main.rs:5:43:
+no jobserver from env: NotPresent
+[NOTE] run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+thread 'main' panicked at src/main.rs:6:17:
+assertion failed: status.success()
+[NOTE] run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+...
```

Comment:
globbing the error line.
 
https://github.com/rust-lang/cargo/actions/runs/9803335785/job/27069355328#step:11:4063
```
 running `D:\a\cargo\cargo\target\debug\cargo.exe run -j2 --config 'target.x86_64-pc-windows-msvc.runner = "runner"'`
thread 'jobserver::runner_inherits_jobserver' panicked at tests\testsuite\jobserver.rs:256:10:
---- expected: tests\testsuite\jobserver.rs:244:27
++++ actual:   stderr
   1    1 | [FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
   2    2 | [RUNNING] `runner target/debug/cargo-jobserver-check[EXE]`
   3    3 | this is a runner
   4    4 | thread 'main' panicked at src/main.rs:5:43:
   5    5 | no jobserver from env: NotPresent
   6    6 | [NOTE] run with `RUST_BACKTRACE=1` environment variable to display a backtrace
   7    7 | thread 'main' panicked at src/main.rs:6:17:
   8    8 | assertion failed: status.success()
   9    9 | [NOTE] run with `RUST_BACKTRACE=1` environment variable to display a backtrace
       10 + [ERROR] process didn't exit successfully: `runner target/debug/cargo-jobserver-check[EXE]` ([EXIT_STATUS]: 101)
```